onnxruntime/core/session/inference_session.cc

```diff
@@ -220,11 +222,73 @@ common::Status InferenceSession::Load(std::function<common::Status(std::shared_p
   return status;
 }
 
+std::shared_ptr<SchemaRegistryManager> InferenceSession::CreateSchemaRegistryManager()
+{
+  auto schema_registry_manager = std::make_shared<SchemaRegistryManager>();
+  for (const auto& schema_collection : custom_schema_registries_) {
+    schema_registry_manager->RegisterRegistry(schema_collection);
+  }
+  auto new_registry_func = [&] (void* info) {
+    auto node = reinterpret_cast<Node*>(info);
+    if (node->OpType() == "PyOp") {
+      OnnxAttrs attrs;
+      OnnxTypes input_types, output_types;
+      std::string module, class_name, compute = "compute", domain = node->Domain();
+
+      for (const auto& iter: node->GetAttributes()) {
+
+          if (iter.second.has_s()) {
+              if      (iter.first == "module")      module            = iter.second.s();
+              else if (iter.first == "class_name")  class_name        = iter.second.s();
+              else if (iter.first == "compute")     compute           = iter.second.s();
+              else                                  attrs[iter.first] = iter.second.s();
+          } else if (iter.second.ints_size() > 0) {
+              if (iter.first == "input_types") {
+                  for (int ii = 0; ii < iter.second.ints_size(); ++ii) {
+                      input_types.push_back(static_cast<ONNXTensorElementDataType>(iter.second.ints(ii)));
+                  }
+              } else if (iter.first == "output_types") {
+                  for (int ii = 0; ii < iter.second.ints_size(); ++ii) {
+                      output_types.push_back(static_cast<ONNXTensorElementDataType>(iter.second.ints(ii)));
+                  }
+              }
+          } else {
+              LOGS(*session_logger_, WARNING) << "PyOp only accept string or tensorproto attribute";
+          }
+      }//for
+
+      ORT_ENFORCE (    module != "", "PyOp module not specified");
+      ORT_ENFORCE (class_name != "", "PyOp class name not specified");
+
+      ORT_ENFORCE (!input_types.empty() , "PyOp node inputs not specified");
+      ORT_ENFORCE (!output_types.empty(), "PyOp node outputs not specified");
+
+      ORT_ENFORCE ( input_types.size() == node->InputDefs().size() , "PyOp node input types mismatch with inputs");
+      ORT_ENFORCE (output_types.size() == node->OutputDefs().size(), "PyOp node output types mismatch with outputs");
+
+      auto& warning_logger = *session_logger_;
+      auto pyop = std::make_unique<PyCustomOp>(attrs, input_types, output_types, module, class_name, compute,
+                                               [warning_logger] (const char* msg) { LOGS(warning_logger, WARNING) << msg; });
+      auto pyop_domain = OrtCreateCustomOpDomain(domain.c_str());
+      ORT_THROW_ON_ERROR(OrtCustomOpDomain_Add(pyop_domain, pyop.get()));
+      AddCustomOpDomains({pyop_domain});
+      pyops_.push_back(std::move(pyop));
+      pyop_domains_.push_back(std::unique_ptr<OrtCustomOpDomain>(pyop_domain));
+      return custom_schema_registries_.back();
```

Comment:
Why back()?

Reply:
First, AddCustomOpDomain will append a new registry into custom_schema_registries.
Next, this new registry must be shared with the graph where the PyOp node is found and add to its own registries, that's why we are returning it to the caller, namely, the current graph.